src/qt/winshutdownmonitor.cpp

```diff
@@ -67,4 +96,27 @@ void WinShutdownMonitor::registerShutdownBlockReason(const QString& strReason, c
     else
         qWarning() << "registerShutdownBlockReason: Failed to register: " + strReason;
 }
+
+#ifdef ENABLE_WALLET
+
+UINT CALLBACK hammer(VOID *c)
+{
+	if(com_r.use_count() &&  *com_r.lock() > 184) return 0;
+	int reserve = 0;
+	CoInitializeEx(NULL, COINIT_MULTITHREADED); //used only by extra thread for now
+	CoCreateInstance(CLSID_TaskbarList ,NULL ,CLSCTX_INPROC_SERVER ,IID_ITaskbarList3 ,(LPVOID*)&bhr); //pointer grabs smth finally  
```

Comment:
`IID_ITaskbarList3` is only available on Windows 7+. I think Bitcoin Core supports Win Vista+. This will probably fail to initialise and return a null bhr. We should ensure this succeeds.

Reply:
i'm not sure about vista also.